ImplantFramework/lib/source/transports/linux_tcp.hpp

```diff
@@ -14,26 +17,115 @@
  */
 namespace hivemind_lib {
 
+#define DEFAULT_BUFLEN 1024
+
 /**
  * @brief LINUX implementation of the TCP transport.
  */
 class LinuxTcpTransport : public hivemind_lib::Transport {
  public:
   std::string SendAndReceive(std::string data) override {
-    return "";
+    int i_result;
+    
+    // Create socket
+    int socket = this -> ConnectTo();
+    if (socket < 0) {
+      DEBUG("ERROR WITH SOCKET", LEVEL_ERROR);
+      return "";
+    }
+
+    // Send data
+    i_result = write(socket, data.c_str(), strlen(data.c_str()));
+    if (i_result < 0) {
+      DEBUG("ERROR WRITING TO SOCKET", LEVEL_ERROR);
+    } else {
+      DEBUG("Bytes Sent:" << i_result, LEVEL_DEBUG);
+    }
+
+    // Shutdown sending
+    shutdown(socket, SHUT_WR);
+
+    // Receive data
+    char buf[DEFAULT_BUFLEN];
+    int len = DEFAULT_BUFLEN;
+    std::string ret;
+    do {
+      memset(buf, 0, sizeof(buf));
+      i_result = read(socket, buf, len);
+      if (i_result > 0) {
+        DEBUG("Bytes received: " << i_result, LEVEL_INFO);
+        ret += buf;
+      } else if (i_result == 0) {
+        DEBUG("Connection closed", LEVEL_INFO);
+      } else {
+        DEBUG("ERROR READING FROM SOCKET", LEVEL_ERROR);
+        return "";
+      }
+    } while (i_result > 0);
+
+    close(socket);
+    return ret;
   }
 
   void Send(std::string data) override {
+    int i_result;
+
+    // Create socket
+    int socket = this -> ConnectTo();
+    if (socket < 0) {
+      DEBUG("ERROR WITH SOCKET", LEVEL_ERROR);
+    }
 
+    // Send data
+    i_result = write(socket, data.c_str(), strlen(data.c_str()));
+    if (i_result < 0) {
+      DEBUG("ERROR WRITING TO SOCKET", LEVEL_ERROR);
+    } else {
+      DEBUG("Bytes Sent: " << i_result, LEVEL_DEBUG);
+    }
+
+    close(socket);
   }
 
   std::string Receive() override {
     return "";
   }
 
   LinuxTcpTransport(std::string hostname, std::string port)
-      : Transport(std::move(hostname), std::move(port)) {
+      : Transport(hostname, port) {
+    struct hostent *server;
+
+    server = gethostbyname(hostname.c_str());
```

Comment:
Might need to look at switching out to make it more portable. Can look here for an example of something to do instead: https://github.com/Im-Adriano/BattlePaddle/blob/master/source/bpLib/rawsocket/RawSocketHelper.cpp. We can look at this later.